.github/workflows/centraldb_angular_docker_publish.yaml

```diff
@@ -49,8 +50,6 @@ jobs:
       id: version
       if: steps.filter.outputs.version == 'true'
       run: |
-        docker tag ${{env.IMG}}:${TAG} ${{env.IMG}}:${VERSION_TAG}
+        export VERSION_TAG=$(cat releasing/version/VERSION)
+        docker tag ${{env.IMG}}:${{env.TAG}} ${{env.IMG}}:${VERSION_TAG}
         docker push ${{env.IMG}}:${VERSION_TAG}
```

Comment:
Let's re-use the makefile rules here, to avoid duplication of code in makefiles and GH Actions. The layers are already cached so rebuilding the image should be very fast
```bash
export TAG=$(cat releasing/version/VERSION)
cd components/centraldashboard-angular
make docker-build docker-push
```

Reply:
Sure! However, we can't avoid duplication of code for building/pushing the notebook server images. This is because we don't have an `IMG` env var in the Makefiles (e.g [codeserver-python Makefile](https://github.com/kubeflow/kubeflow/blob/658f5b8a76e3b19438678decaf21a9c2fdfd9af1/components/example-notebook-servers/codeserver-python/Makefile#L1)) to set the `registry` there. Therefore with the current Makefiles we have to use:
- `make docker-build`
- `docker tag` to add the registry
- `and finally docker push`

For example:
```yaml
- name: Build and push Notebook Server images
      run: |
        export TAG=$(shell git describe --tags --always --dirty)
        cd components/example-notebook-servers/
        make docker-build -C codeserver-python TAG=${TAG}
        docker tag codeserver-python:${TAG} ${{env.REGISTRY}}/codeserver-python:${TAG}
        docker push ${{env.REGISTRY}}/codeserver-python:${TAG}

    - name: Build and push latest Notebook Server images
      if: github.ref == 'refs/heads/master'
      run: |  
        export TAG=latest
        cd components/example-notebook-servers/
        make docker-build -C codeserver-python TAG=${TAG}
        docker tag codeserver-python:${TAG} ${{env.REGISTRY}}/codeserver-python:${TAG}
        docker push ${{env.REGISTRY}}/codeserver-python:${TAG}

    - name: Build and push Notebook Server images on Version change
      id: version
      if: steps.filter.outputs.version == 'true'
      run: |
        export TAG=$(cat releasing/version/VERSION)
        cd components/example-notebook-servers/
        make docker-build -C codeserver-python TAG=${TAG}
        docker tag codeserver-python:${TAG} ${{env.REGISTRY}}/codeserver-python:${TAG}
        docker push ${{env.REGISTRY}}/codeserver-python:${TAG}
```

Reply:
@apo-ger let's leave the notebook images completely out for this effort.

I believe we would be able to use the Makefiles if we'd use a `REGISTRY` env var in each Makefile. But let's discuss this in a follow up PR, since this one is already getting big

Reply:
@kimwnasptd 
> let's leave the notebook images completely out for this effort.
Should I remove the logic for building with latest tag or remove the notebook-server workflows completely and tackle them in the follow up PR?

Reply:
just the changes from this PR, and we can send another PR for the latest tag in a separate PR that:
1. Uses a `REGISTRY` env var in all makefiles (let's see if we need to do more)
2. Use the Makefiles to build each notebook, but with a different TAG each time like we do for the rest of the components